### DISCUSSION TITLE
How to split dataset into train and valid?

### DISCUSSION THREAD

**Discussion by jexterliangsufe** [Nixtla/neuralforecast#1042] [Q&A]
My train dataloader length is equal to valid dataloader length. Why? 
In nf codes, L329 create self.val_size, but not used in creating datamodule L336 
https://github.com/Nixtla/neuralforecast/blob/90b6fd06dc59fa3f320c57f361d4ca8379aa6be7/neuralforecast/common/_base_model.py#L329-L343
Then pytorch_lightning.Trainer is called in L357 and does not use val_size if is_local=True
https://github.com/Nixtla/neuralforecast/blob/90b6fd06dc59fa3f320c57f361d4ca8379aa6be7/neuralforecast/common/_base_model.py#L357
Also, In TimeSeriesDataModule, the function train_dataloader and val_dataloader use the same dataset(both self.dataset). 
https://github.com/Nixtla/neuralforecast/blob/90b6fd06dc59fa3f320c57f361d4ca8379aa6be7/neuralforecast/tsdataset.py#L394-L412
Is this a mistake or I use nf in a wrong way?
- ----
neuralforecast=1.7.2
pytorch_lightning=2.2.5

**Comment by jexterliangsufe** [comment]
I made a mistake. `self.val_size` used in _base_windows.py and split train/valid/predict windows correctly.

**Reply by samosun** [reply]
Hi, I have the same confusion. Correct me if I'm wrong,  is the train/validation set divided on each batch?

**Reply by jexterliangsufe** [reply]
Yes.
